app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt

```diff
@@ -0,0 +1,46 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun FavoriteButton(
+    modifier: Modifier = Modifier
+) {
+
+    var isFavorite by rememberSaveable { mutableStateOf(true) }
```

Comment:
This state should be hoisted. The idea is that we want to pass logic from ViewModel. This logic should be passed from a top level component which knows what to do when the state changes. 
In your case this will work fine, but afterwards we will add some extra logic on like button press (for example we will save that state somehow (maybe with a database) so we have all our liked movies when we re-enter application. We don't want to have that (saving) logic in View component, that should be placed in Use case which will be called by ViewModel. All of this will be explained in class but for now be aware that should always be hoisted. 
Hint: your FavoriteButton composable should have these params


```
@Composable
fun FavoriteButton(
    isFavorite: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    ....
}
```

Read more about state hoisting here : https://developer.android.com/jetpack/compose/state